src/openvslam_ros.cc

```diff
@@ -45,6 +48,51 @@ void system::publish_pose() {
     pose_msg.pose.pose.position.y = trans(1);
     pose_msg.pose.pose.position.z = trans(2);
     pose_pub_->publish(pose_msg);
+
+    // Send map->odom transform. Set publish_tf_ to false with not using TF
+    if(publish_tf_){
+
+        tf2::Stamped<tf2::Transform> camera_to_map(tf2::Transform(tf2::Quaternion(quat.x(), quat.y(), quat.z(), quat.w()), 
+                                                    tf2::Vector3(trans(0), trans(1), trans(2))).inverse(),
+                                                    tf2_ros::fromMsg(node_->now()), camera_link_);
+
+        geometry_msgs::msg::TransformStamped camera_to_map_msg, odom_to_map_msg, map_to_odom_msg;
+        tf2::Stamped<tf2::Transform> odom_to_map_stamped, map_to_odom_stamped;
+
+        // camera_to_map_msg = tf2::toMsg(camera_to_map); - it breaks the execution
+        camera_to_map_msg.header.stamp = tf2_ros::toMsg(camera_to_map.stamp_);
+        camera_to_map_msg.header.frame_id = camera_to_map.frame_id_;
+        camera_to_map_msg.transform.translation.x = camera_to_map.getOrigin().getX();
+        camera_to_map_msg.transform.translation.y = camera_to_map.getOrigin().getY();
+        camera_to_map_msg.transform.translation.z = camera_to_map.getOrigin().getZ();
+        camera_to_map_msg.transform.rotation = tf2::toMsg(camera_to_map.getRotation());
+        
+        try {
+            odom_to_map_msg = tf_->transform(camera_to_map_msg, odom_frame_);
+            tf2::fromMsg(odom_to_map_msg, odom_to_map_stamped);
+            map_to_odom_ = tf2::Transform(tf2::Quaternion(odom_to_map_stamped.getRotation()), tf2::Vector3(odom_to_map_stamped.getOrigin())).inverse();
```

Comment:
I do not see that `map_to_odom_` variable is used anywhere, except of `publish_pose()`. So, this could be a local variable instead of class' field. Setting the identity to it is unnecessary in that case.